src/sagemaker_server.cc

```diff
@@ -628,6 +628,68 @@ SagemakerAPIServer::SageMakerMMEHandleInfer(
   }
 }
 
+bool
+SagemakerAPIServer::SageMakerMMEUnloadModelCheckStatus(const char* model_name)
+{
+  /* Use the RepositoryIndex API to check if the model state has become
+  UNAVAILABLE i.e. model is no longer in the 'in-the-process-of' being
+  UNLOADED. Consequently, the reason field should be 'unloaded'.*/
+  TRITONSERVER_Message* server_model_index_message_ = nullptr;
+  uint32_t ready_flag = 0;  // value of 1 should be set if only the 'ready'
+                            // models are required from the index. In this case,
+                            // we need all models.
+  TRITONSERVER_ServerModelIndex(
+      server_.get(), ready_flag, &server_model_index_message_);
+
+  std::shared_ptr<TRITONSERVER_Message> managed_msg(
+      server_model_index_message_,
+      [](TRITONSERVER_Message* msg) { TRITONSERVER_MessageDelete(msg); });
+
+  const char* index_buffer;
+  size_t index_byte_size;
+
+  /*Handle the return from this function correctly*/
+  TRITONSERVER_MessageSerializeToJson(
+      server_model_index_message_, &index_buffer, &index_byte_size);
+
+  /* Read into json buffer*/
+  triton::common::TritonJson::Value server_model_index_json;
+  server_model_index_json.Parse(index_buffer, index_byte_size);
+
+  const char* name;
+  size_t name_len;
+
+  const char* version;
+  size_t version_len;
+
+  const char* state;
+  size_t state_len;
+
+  const char* reason;
+  size_t reason_len;
+
+  bool is_model_unavailable = false;
+
+
+  for (size_t id = 0; id < server_model_index_json.ArraySize(); ++id) {
+    triton::common::TritonJson::Value index_json;
+    server_model_index_json.IndexAsObject(id, &index_json);
+
+    index_json.MemberAsString("name", &name, &name_len);
+    index_json.MemberAsString("version", &version, &version_len);
```

Comment:
Do you need to match on a particular version? Or with sagemaker API, is there only ever one version for a given model name?

Reply:
That's correct, we expect only one version of for a model name